methods/basic.go

```diff
@@ -15,7 +20,62 @@ func BasicRequestWithBody(c *cli.Context, method string) (string, error) {
 		return "", err
 	}
 
-	var jsonStr = []byte(c.String("body"))
+	var jsonStr []byte
+
+	if c.Bool("editor") {
+		var path string
+		var editor string
+
+		// check OS
+		currentOs := runtime.GOOS
+
+		// based on OS find assign default editor
+		if currentOs == "linux" || currentOs == "darwin" {
+			editor = "nano"
+		} else if currentOs == "windows" {
+			editor = "notepad"
+		}
+
+		path, err = exec.LookPath(editor)
+		if err != nil {
+			return "", fmt.Errorf("Unable to locate %s at %s", editor, path)
+		}
+
+		// create temp file for writing request body
+		tempFile, err := ioutil.TempFile("", "HOPP-CLI-REQUEST*.txt")
+		if err != nil {
+			return "", fmt.Errorf("Unable to create temp file: %s\n%s", tempFile.Name(), err)
+		}
+
+		tempFileName := string(tempFile.Name())
+		tempFile.Close()
+
+		// open temp file in selected editor and wait until closed
+		cmd := exec.Command(path, tempFileName)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+
+		err = cmd.Start()
+		if err != nil {
+			return "", fmt.Errorf("Unable to open editor: %s", err)
+		}
+
+		color.Yellow("Waiting for file to close..\n\n")
+		cmd.Wait()
+
+		// read temp file contents
+		fileData, err := ioutil.ReadFile(tempFileName)
+		if err != nil {
+			return "", fmt.Errorf("Error reading file: %s", err)
+		}
+
+		jsonStr = []byte(string(fileData))
```

Comment:
this is a conversion from []byte->string->[]byte, which isn't needed. Since `err` has already been declared as well as `jsonStr`, you can assign directly without the initializer shorthand, like:

```go
jsonStr, err = ioutil.ReadFile(tempFileName)
```